dot/core/service.go

```diff
@@ -551,3 +566,30 @@ func (s *Service) GetReadProofAt(block common.Hash, keys [][]byte) (
 
 	return block, proofForKeys, nil
 }
+
+// getRuntimeFromDB gets the runtime for the corresponding block hash from storageState
+func (s *Service) getRuntimeFromDB(blockHash *common.Hash) (instance runtime.Instance, err error) {
+	var stateRootHash *common.Hash
+	if blockHash != nil {
+		stateRootHash, err = s.storageState.GetStateRootFromBlock(blockHash)
+		if err != nil {
+			return nil, fmt.Errorf("getting state root from block hash: %w", err)
+		}
+	}
```

Comment:
I would like to suggest we accept a `blockHash common.Hash` argument and not a `nil` hash.
We do this all over the codebase and it's really dark magic at the end of the day, which we should avoid in new code (until we refactor older code one day). Anyway I don't think we pass in `nil` anywhere so it shouldn't be too troublesome to change.